### PR TITLE
Migrate to Ballerina 1.0.0

### PR DESCRIPTION
**Pull request by rosensilva**

### CODE REVIEW

File: Ballerina.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [project]
 org-name = "wso2"
-version = "0.9.20"
-authors = ["wso2"]
+version = "0.10.20"
```
Comment (Asitha) [inline Ballerina.toml]:
```suggestion
version = "0.10.0"
```

File: src/scim2/Module.md [hunk 2]
```diff
@@ -29,34 +29,28 @@ import wso2/scim2;
 Instantiate the connector by giving authentication details in the HTTP client config. The HTTP client config has built-in support for BasicAuth and OAuth 2.0. The SCIM2 connector can be minimally instantiated using the access token or using the client ID, client secret, and refresh token in the HTTP client config or BasicAuth configuration.
 
 ```ballerina
-scim2:Scim2Configuration scim2Config = {
-    url:url,
+Scim2Configuration scim2Config = {
```
Comment (Maninda) [inline src/scim2/Module.md]:
```suggestion
scim2:Scim2Configuration scim2Config = {
```
Should contain the package.

File: src/scim2/Module.md [hunk 3]
```diff
@@ -29,34 +29,28 @@ import wso2/scim2;
...
+        trustStore: {
+            path: config:getAsString("KEYSTORE"),
+            password: config:getAsString("KEYSTORE_PASSWORD")
         }
     }
+    
 };
 
-scim2:Client scimEP = new(scim2Config);
+Client scimEP = new(scim2Config);
```
Comment (Maninda) [inline src/scim2/Module.md]:
```suggestion
scim2:Client scimEP = new(scim2Config);
```
Should contain the package.

File: src/scim2/scim2_data_mapping.bal [hunk 4]
```diff
@@ -26,12 +26,14 @@ function convertJsonToGroup(json sourceJsonObject) returns Group {
 }
 
 
-function convertJsonToMeta(json sourceJsonObject) returns Meta {
+function convertJsonToMeta(json | error sourceJsonObject) returns Meta {
```
Comment (Maninda) [inline src/scim2/scim2_data_mapping.bal]:
```suggestion
function convertJsonToMeta(json|error sourceJsonObject) returns Meta {
```
No space is used in convention.

File: src/scim2/scim2_data_mapping.bal [hunk 5]
```diff
@@ -76,98 +80,105 @@ function toListMem(Group g) returns json[] {
...
+        postalCode: sourceAddressStruct.postalCode != "" ? sourceAddressStruct.postalCode : "",
+        primary: sourceAddressStruct.primary != "" ? sourceAddressStruct.primary : "",
+        region: sourceAddressStruct.region != "" ? sourceAddressStruct.region : "",
+        'type: sourceAddressStruct.'type != "" ? sourceAddressStruct.'type : ""
+    };
     return targetJsonObject;
 }
 
-function convertJsonToName(json sourceJsonObject) returns Name {
+function convertJsonToName(json | error sourceJsonObject) returns Name {
```
Comment (Maninda) [inline src/scim2/scim2_data_mapping.bal]:
```suggestion
function convertJsonToName(json|error sourceJsonObject) returns Name {
```
No space in convntion

File: src/scim2/scim2_data_mapping.bal [hunk 6]
```diff
@@ -177,59 +188,66 @@ function convertJsonToCertificate(json sourceJsonObject) returns X509Certificate
     return targetCertificate;
 }
 
-function convertJsonToEnterpriseExtension(json sourceJsonObject) returns EnterpriseUserExtension {
+function convertJsonToEnterpriseExtension(json | error sourceJsonObject) returns EnterpriseUserExtension {
```
Comment (Maninda) [inline src/scim2/scim2_data_mapping.bal]:
```suggestion
function convertJsonToEnterpriseExtension(json|error sourceJsonObject) returns EnterpriseUserExtension {
```
No space in convention

File: src/scim2/scim2_data_mapping.bal [hunk 7]
```diff
@@ -177,59 +188,66 @@ function convertJsonToCertificate(json sourceJsonObject) returns X509Certificate
...
+        targetEnterpriseUser.organization = sourceJsonObject.organization != null ?
+        sourceJsonObject.organization.toString() : "";
+        targetEnterpriseUser.manager = sourceJsonObject.manager != null ?
+        convertJsonToManager(sourceJsonObject.manager) : {};
+    }
     return targetEnterpriseUser;
 }
 
-function convertJsonToManager(json sourceJsonObject) returns Manager {
+function convertJsonToManager(json | error sourceJsonObject) returns Manager {
```
Comment (Maninda) [inline src/scim2/scim2_data_mapping.bal]:
```suggestion
function convertJsonToManager(json|error sourceJsonObject) returns Manager {
```

File: src/scim2/scim2_endpoint.bal [hunk 8]
```diff
@@ -0,0 +1,782 @@
...
+            }
+        } else {
+            error err = error(SCIM2_ERROR_CODE, message = "Error occurred while invoking the SCIM2 API.");
+            return err;
+        }
+    }
+
+    # Returns a list of group records if found or error if any error occured.
+    # + return - If success, returns list of Group objects, else returns error
+    public remote function getListOfGroups() returns @tainted (Group[] | error) {
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]:
```suggestion
    public remote function getListOfGroups() returns @tainted (Group[]|error) {
```

File: src/scim2/scim2_endpoint.bal [hunk 9]
```diff
@@ -0,0 +1,782 @@
...
+        } else {
+            error err = error(SCIM2_ERROR_CODE, message = "Error occurred while invoking the SCIM2 API");
+            return err;
+        }
+    }
+
+    # Returns a group record with the specified group name if found.
+    # + groupName - Name of the group
+    # + return - If success, returns Group object, else returns error
+    public remote function getGroupByName(string groupName) returns (Group | error) {
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]:
```suggestion
    public remote function getGroupByName(string groupName) returns (Group|error) {
```

File: Ballerina.toml (hunk 1 above)
Comment (rosensilva) [reply Ballerina.toml]: fixed

File: src/scim2/Module.md (hunk 3 above)
Comment (rosensilva) [reply src/scim2/Module.md]: fixed

File: src/scim2/scim2_endpoint.bal [hunk 10]
```diff
@@ -0,0 +1,782 @@
...
+            var receivedUser = resolveUser(userName, resUser);
+            if (receivedUser is User) {
+                user = receivedUser;
+                string userId = user.id;
+                var res = self.scimClient->delete(SCIM_USER_END_POINT + SCIM_FILE_SEPERATOR + userId, request);
+                if (res is http:Response) {
+                    if (res.statusCode == HTTP_NO_CONTENT) {
+                        return "deleted";
+                    } else {
+                        error Error = error(SCIM2_ERROR_CODE, message = failedMessage + res.reasonPhrase);
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]: Capital variable name _Error_ is used. B7A variable names are in Camel Case in convention.

File: src/scim2/scim2_endpoint.bal [hunk 11]
```diff
@@ -0,0 +1,782 @@
...
+        }
+
+        http:Request request = new ();
+
+        json[] emailList = [];
+        json email;
+        int i = 0;
+        foreach var emailAddress in emails {
+            if (!stringutils:equalsIgnoreCase(emailAddress.'type, SCIM_WORK) && !stringutils:equalsIgnoreCase(emailAddress.'type, SCIM_HOME)) {
+                error Error = error(SCIM2_ERROR_CODE, message = "Email type should be defiend as either home or work");
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]: Capital _Error_

File: src/scim2/scim2_endpoint.bal [hunk 12]
```diff
@@ -0,0 +1,782 @@
...
+
+        request = createRequest(body);
+
+        string url = SCIM_USER_END_POINT + SCIM_FILE_SEPERATOR + id;
+        var res = self.scimClient->patch(url, request);
+        if (res is http:Response) {
+            if (res.statusCode == HTTP_OK) {
+                return "Email updated";
+            } else {
+                error Error = error(SCIM2_ERROR_CODE, message = res.reasonPhrase);
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]: Capital _Error_

File: src/scim2/Module.md (hunk 2 above)
Comment (rosensilva) [reply src/scim2/Module.md]: fixed

File: src/scim2/scim2_endpoint.bal [hunk 13]
```diff
@@ -0,0 +1,782 @@
...
+        }
+    }
+
+    # Update addresses of a user.
+    # + id - ID of the user
+    # + addresses - List of new addresses of the user
+    # + return - If success, returns string message with status, else returns error
+    public remote function updateAddresses(string id, Address[] addresses) returns (string | error) {
+        if (stringutils:equalsIgnoreCase(id, "")) {
+            error Error = error(SCIM2_ERROR_CODE, message = "User should be valid");
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]: Capital _Error_

File: src/scim2/scim2_endpoint.bal [hunk 14]
```diff
@@ -0,0 +1,782 @@
...
+
+        json body = convertUserToJson(user, "update");
+        request = createRequest(body);
+        string url = SCIM_USER_END_POINT + SCIM_FILE_SEPERATOR + user.id;
+        var res = self.scimClient->put(url, request);
+        if (res is http:Response) {
+            if (res.statusCode == HTTP_OK) {
+                return "User updated";
+            } else {
+                error Error = error(SCIM2_ERROR_CODE, message = res.reasonPhrase);
```
Comment (Maninda) [inline src/scim2/scim2_endpoint.bal]: Capital _Error_

File: src/scim2/scim2_utils.bal [hunk 15]
```diff
@@ -16,12 +15,13 @@
...
 //
 import ballerina/http;
+import ballerina/mime;
 
 # Returns a user record if the input http:Response contains a user.
 # + userName - User name of the user
 # + response - http:Response with the received response from the SCIM2 API
 # + return - If success returns User object, else returns error
-function resolveUser(string userName, http:Response response) returns User|error {
+function resolveUser(string userName, http:Response response) returns @untainted User | error {
```
Comment (Maninda) [inline src/scim2/scim2_utils.bal]:
```suggestion
function resolveUser(string userName, http:Response response) returns @untainted User|error {
```

File: src/scim2/scim2_utils.bal [hunk 16]
```diff
@@ -54,7 +54,7 @@ function resolveUser(string userName, http:Response response) returns User|error
 # + groupName - Name of the group
 # + response - http:Response with the received response from the SCIM2 API
 # + return - If success returns Group object, else returns error
-function resolveGroup(string groupName, http:Response response) returns Group|error {
+function resolveGroup(string groupName, http:Response response) returns @untainted Group | error {
```
Comment (Maninda) [inline src/scim2/scim2_utils.bal]:
```suggestion
function resolveGroup(string groupName, http:Response response) returns @untainted Group|error {
```

File: src/scim2/scim2_utils.bal [hunk 17]
```diff
@@ -96,54 +96,56 @@ function createRequest(json body) returns http:Request {
 # + valueType - The name of the user attribute
 # + newValue - The new value of the attribute
 # + return - If success returns `json` object, else returns error
-function createUpdateBody(string valueType, string newValue) returns json|error {
+function createUpdateBody(string valueType, string newValue) returns json | error {
```
Comment (Maninda) [inline src/scim2/scim2_utils.bal]:
```suggestion
function createUpdateBody(string valueType, string newValue) returns json|error {
```

File: src/scim2/scim2_endpoint.bal (hunk 10 above)
Comment (rosensilva) [reply src/scim2/scim2_endpoint.bal]: Fixed

File: src/scim2/scim2_endpoint.bal (hunk 11 above)
Comment (rosensilva) [reply src/scim2/scim2_endpoint.bal]: Fixed

File: src/scim2/scim2_endpoint.bal (hunk 12 above)
Comment (rosensilva) [reply src/scim2/scim2_endpoint.bal]: Fixed

File: src/scim2/scim2_endpoint.bal (hunk 14 above)
Comment (rosensilva) [reply src/scim2/scim2_endpoint.bal]: Fixed

File: src/scim2/scim2_endpoint.bal (hunk 13 above)
Comment (rosensilva) [reply src/scim2/scim2_endpoint.bal]: fixed

**Pull request merged**